src/utils/Metadata.js

```diff
@@ -0,0 +1,22 @@
+const TimeInterval = require("./TimeInterval.js");
+
+class Metadata {
+    timeInterval;
+    initEndpoint;
+    lastEndpoint;
```

Comment:
lastEndpoint is not clear, you should use lastUsedEndpoint or lastDownloadedEndpoint